src/compiler/legend.ts

```diff
@@ -58,9 +60,12 @@ export function compileLegend(model: Model, channel: Channel, def) {
 }
 
 export function title(model: Model, channel: Channel) {
-  let leg = model.fieldDef(channel).legend;
+  // https://github.com/Microsoft/TypeScript/issues/5842
+  const legend: any = model.fieldDef(channel).legend;
 
-  if (leg.title) return leg.title;
+  if (isLegend(legend) && legend.title) {
```

Comment:
`legend` is of type `Legend | boolean` so it does not necessarily have a property `title`.